api/src/services/user-groups/create-service.ts

```diff
@@ -25,14 +26,34 @@ export class CreateService extends BaseService {
       throw new Error("Group ID is required")
     }
 
-    const userGroup = await UserGroup.create({
-      creatorId: this.currentUser.id,
-      userId,
-      groupId,
-      ...optionalAttributes,
+    return db.transaction(async () => {
+      // _TODO_ can I eager load the user and group?
+      const userGroup = await UserGroup.create({
+        creatorId: this.currentUser.id,
+        userId,
+        groupId,
+        ...optionalAttributes,
+      })
+
+      const user = await User.findByPk(userId)
+      const group = await Group.findByPk(groupId)
+
+      if (isNil(user)) {
+        throw new Error("User not found")
+      }
+
+      if (isNil(group)) {
+        throw new Error("Group not found")
+      }
+
+      await Notifications.Groups.NotifyUserOfMembershipService.perform(
+        user,
+        group,
+        this.currentUser
+      )
```

Comment:
Consider moving this into a local private method.
e.g.
```ts
this.notifyUserofMembershipService(userId, groupId, this.currentUser)
```
so that the main "perform" method reads like a list of actions, rather than the implementation of those actions.